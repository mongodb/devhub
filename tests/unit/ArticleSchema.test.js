import React from 'react';
import { shallow } from 'enzyme';
import ArticleSchema from '../../src/components/dev-hub/article-schema';

import mockData from './data/ArticleSchema.test.json';

describe('ArticleSchema', () => {
    let shallowWrapper;

    beforeAll(() => {
        shallowWrapper = shallow(<ArticleSchema {...mockData} />);
    });

    it('script has a correct schema', () => {
        const script = shallowWrapper.find('script');

        expect(script.text()).toEqual(
            JSON.stringify({
                '@context': 'https://schema.org',
                '@type': 'BlogPosting',
                url: mockData.articleUrl,
                headline: mockData.title,
                description: mockData.description,
                datePublished: mockData.publishedDate,
                dateModified: mockData.modifiedDate,
                mainEntityOfPage: {
                    '@type': 'WebPage',
                    url: mockData.articleUrl,
                },
                image: {
                    '@type': 'imageObject',
                    url: mockData.imageUrl,
                },
                publisher: {
                    '@type': 'Organization',
                    name: 'MongoDB',
                    logo: {
                        '@type': 'imageObject',
                        url: 'https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png',
                    },
                },
                author: [
                    {
                        '@type': 'Person',
                        name: mockData.authors[0].name,
                    },
                ],
                tags: ['atlas', 'mongodb'],
                inLanguage: 'English',
            })
        );

        expect(shallowWrapper).toMatchSnapshot();
    });

    it('script has a correct schema without modified date', () => {
        shallowWrapper = shallow(
            <ArticleSchema
                {...mockData}
                modifiedDate={undefined}
                language="Test"
            />
        );
        const script = shallowWrapper.find('script');

        expect(script.text()).toEqual(
            JSON.stringify({
                '@context': 'https://schema.org',
                '@type': 'BlogPosting',
                url: mockData.articleUrl,
                headline: mockData.title,
                description: mockData.description,
                datePublished: mockData.publishedDate,
                dateModified: mockData.publishedDate,
                mainEntityOfPage: {
                    '@type': 'WebPage',
                    url: mockData.articleUrl,
                },
                image: {
                    '@type': 'imageObject',
                    url: mockData.imageUrl,
                },
                publisher: {
                    '@type': 'Organization',
                    name: 'MongoDB',
                    logo: {
                        '@type': 'imageObject',
                        url: 'https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png',
                    },
                },
                author: [
                    {
                        '@type': 'Person',
                        name: mockData.authors[0].name,
                    },
                ],
                tags: ['atlas', 'mongodb'],
                inLanguage: 'Test',
            })
        );

        expect(shallowWrapper).toMatchSnapshot();
    });
});
