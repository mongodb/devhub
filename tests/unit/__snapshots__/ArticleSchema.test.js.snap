// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleSchema script has a correct schema 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    id="structured-data"
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"BlogPosting","url":"https://www.mongodb.com/developer/how-to/zap-tweet-repeat-how-to-use-zapier-mongodb/","headline":"Test Title","description":"Test Description","datePublished":"2000-03-03","dateModified":"2000-03-04","mainEntityOfPage":{"@type":"WebPage","url":"https://www.mongodb.com/developer/how-to/zap-tweet-repeat-how-to-use-zapier-mongodb/"},"image":{"@type":"imageObject","url":"test/test.png"},"publisher":{"@type":"Organization","name":"MongoDB","logo":{"@type":"imageObject","url":"https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png"}},"author":[{"@type":"Person","name":"D Test"}],"tags":["atlas","mongodb"],"inLanguage":"English"}
  </script>
</HelmetWrapper>
`;

exports[`ArticleSchema script has a correct schema without modified date 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    id="structured-data"
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"BlogPosting","url":"https://www.mongodb.com/developer/how-to/zap-tweet-repeat-how-to-use-zapier-mongodb/","headline":"Test Title","description":"Test Description","datePublished":"2000-03-03","dateModified":"2000-03-03","mainEntityOfPage":{"@type":"WebPage","url":"https://www.mongodb.com/developer/how-to/zap-tweet-repeat-how-to-use-zapier-mongodb/"},"image":{"@type":"imageObject","url":"test/test.png"},"publisher":{"@type":"Organization","name":"MongoDB","logo":{"@type":"imageObject","url":"https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png"}},"author":[{"@type":"Person","name":"D Test"}],"tags":["atlas","mongodb"],"inLanguage":"Test"}
  </script>
</HelmetWrapper>
`;
